### DISCUSSION TITLE
yarn watch failed

### DISCUSSION THREAD

**Discussion by GracefulTing** [coder/code-server#3488] [Q&A]
When I clone the code and execute yarn && yarn watch, the error is reported as follows，
Want to know how to solve？

 code-server git:(main) ✗ yarn watch
yarn run v1.22.10
$ VSCODE_IPC_HOOK_CLI= NODE_OPTIONS=--max_old_space_size=32384 ts-node ./ci/dev/watch.ts
'VSCODE_IPC_HOOK_CLI' 不是内部或外部命令，也不是可运行的程序
或批处理文件。
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by GracefulTing** [comment]
windows system

**Comment by code-asher** [comment]
Hmm Windows might not like the environment variable syntax. I think we'll need to use cross-env with `yarn add --dev cross-env` and then change the watch command in the `package.json` to:

```
cross-env VSCODE_IPC_HOOK_CLI= NODE_OPTIONS=--max_old_space_size=32384 ts-node ./ci/dev/watch.ts
```

But I tried this and while it works I got stuck on `spawn yarn ENOENT` inside `watch.ts`. I'm not sure why that's happening. Putting the full path to `yarn` doesn't work either.

**Reply by GracefulTing** [reply]
Okay, is there any successful running example for windows users, please provide some advice

**Reply by code-asher** [reply]
Unfortunately there isn't currently an example. Someone will need to test on Windows and figure out how to support it. I created an issue for it here: https://github.com/cdr/code-server/issues/3501

**Comment by GracefulTing** [comment]
After changing to a linux system, the yarn watch error is reported as follows：

[watch-extension-media] error Command failed with exit code 1.
[watch-extension-media] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "watch-client" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Reply by code-asher** [reply]
Could you post more of the log? I think the source of the error might be further up.

**Reply by killmeorgetout** [reply]
PS F:\code\code-server> yarn watch   
yarn run v1.22.17
$ VSCODE_DEV=1 VSCODE_IPC_HOOK_CLI= NODE_OPTIONS='--max_old_space_size=32384 --trace-warnings' ts-node ./ci/dev/watch.ts
'VSCODE_DEV' 不是内部或外部命令，也不是可运行的程序
或批处理文件。
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

I am also windows system, the same problem occurred in `yarn watch`, is there a solution? @code-asher

**Reply by killmeorgetout** [reply]
@GracefulTing did you solved it?

**Reply by jsjoeio** [reply]
Hmm unfortunately I don't think we've added support for development on Windows 😢

I believe right now you can only develop on Linux and macOS.

**Reply by aoyeQ** [reply]
hi, is it currently supported for development on Windows? I'm encountering issues when running 'yarn watch'

$ yarn watch
yarn run v1.22.19
$ cross-env VSCODE_DEV=1 VSCODE_IPC_HOOK_CLI= NODE_OPTIONS=--max_old_space_size=32384 ts-node ./ci/dev/watch.ts
[Compiler][code-server] [18:36:35] Starting compilation in watch mode...
[Code OSS] $ npm-run-all -lp watch-client watch-extensions
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module 'D:\project\code-server\lib\vscode\node_modules\gulp\bin\gulp.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.18.2
error Command failed with exit code 1.
[vscodeWebExtensions] Terminated unexpectedly
[codeServer] Killing...

[vscode] Killing...

[vscodeWebExtensions] Killing...

[plugins] Killing...

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Reply by code-asher** [reply]
There is no official support for Windows right now.  This looks like you just need to run `yarn` though as there are missing modules.
